Use surefire systemPropertyVariables instead of systemProperties

rquinio opened the pull request:
This is a solution for #9339 inside IntelliJ IDE.
Also prepares for upgrading to surefire 3.x Final, as I guess deprecated properties will be removed.

Changes:
- Replace deprecated `systemProperties` by `systemPropertyVariables` everywhere (for both surefire and failsafe plugins)
- Add a warning about Eclipse IDE in logging.adoc
- Remove duplicated surefire configuration of JBoss LogManager in few extensions, as already inherited from build-parent pluginManagement.

rquinio commented on docs/src/main/asciidoc/logging.adoc:
```diff
@@ -239,6 +239,8 @@ In particular, you need to set the appropriate `LogManager` using the `java.util
 <1> Make sure the `org.jboss.logmanager.LogManager` is used.
 <2> Enable debug logging for all logging categories.
 
+WARN: Eclipse IDE will not automatically read Maven surefire configuration, see https://github.com/quarkusio/quarkus/issues/9339
```
Here's the warning

rquinio replied:
Actually asciidoctor proper admonition is WARNING not WARN, I updated it.

gsmet commented on build-parent/pom.xml:
```diff
@@ -216,9 +216,9 @@
-                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                        </systemProperties>
+                        <systemPropertyVariables>
+                          <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
```
This is minor but you broke the indentation.

rquinio replied:
Fixed

gsmet commented on extensions/azure-functions-http/maven-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -80,15 +80,6 @@
-            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-          </systemProperties>
-        </configuration>
-      </plugin>
```
This doesn't look right? I don't see any parent here.

rquinio replied:
Oops, fixed.

gsmet reviewed (changes requested):
Added a few comments.

So Intellij reads the new properties and take them into account but not the old ones? That's why this fix is about?

gsmet commented:
Oh and one more comment for next time: it makes reviews easier if you isolate the massive renamings in separate commits. This way, we can just quickly scan that large commit and focus more on the other changes.

No need to change that but as it seems you want to contribute regularly, I thought I should share that wisdom with you :).

rquinio commented:
> So Intellij reads the new properties and take them into account but not the old ones? That's why this fix is about?

Exactly,  as far as I can tell `systemProperties` has been deprecated since at least 2012, so I guess some tooling didn't even consider it.

gsmet commented:
@rquinio could you have a look at my other comments?

rquinio committed:
Use surefire systemPropertyVariables instead of systemProperties

systemProperties is deprecated in favor of systemPropertyVariables.
This allows to have log levels based on Quarkus configuration when running integration tests from IntelliJ IDE

Command: grep -IRl "systemProperties>" .  --exclude-dir=target | xargs sed -i 's/systemProperties\>/systemPropertyVariables\>/g'

rquinio committed:
Add a warning about surefire and Eclipse IDE in logging.adoc

Remove duplicated surefire configuration in few integration-tests, as already inherited from build-parent

rquinio commented:
> Oh and one more comment for next time: it makes reviews easier if you isolate the massive renamings in separate commits. This way, we can just quickly scan that large commit and focus more on the other changes.
> 
> No need to change that but as it seems you want to contribute regularly, I thought I should share that wisdom with you :).

I've split into 2 commits, since I guess you'd re-review anyway :)
I nailed the sed command: ```grep -IRl "systemProperties>" .  --exclude-dir=target | xargs sed -i 's/systemProperties\>/systemPropertyVariables\>/g'```
I'll do a similar PR on quarkus-quickstarts afterwards.

rquinio commented:
Btw there are 3 more files impacted, but that's because I've also rebased from master, so I picked up the pom.xml from new picocli extension.

gsmet commented:
Thanks!
